### PR TITLE
[feature] `;roleinfo` command 

### LINKED ISSUE #24: ;roleinfo
Gives information such as
- color
- if you have the following permissions (dyno lists useless ones)
  - Audit Log
  - Manage Roles/Server/Channels
  - Kick/Ban
  - Manage messages
  - Mention @ everyone and the such
Status
- high council vs lower council vs citizen

### PR DESCRIPTION
**Pull request by sourTaste000**
Signed-off-by: sourTaste000 <47074495+sourTaste000@users.noreply.github.com>

closes #24

### CODE REVIEW

File: src/main/kotlin/commands/RoleInfoCommand.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import greedyString
+import helpers.StringHelper.toHumanReadable
+import long
+import net.ayataka.kordis.entity.message.Message
+import net.ayataka.kordis.entity.server.role.Role
+
+object RoleInfoCommand : Command("roleinfo") {
+    init {
+        greedyString("roleName") {
+            doesLater {context ->
```
Comment (l1ving) [inline src/main/kotlin/commands/RoleInfoCommand.kt:17]: Require council member to run this

File: src/main/kotlin/commands/RoleInfoCommand.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                val roleName: String = context arg "roleName"
+                val foundRole = message.server?.roles?.findByName(roleName) ?: run {
+                    message.error("Role name not found! Try using the role id.")
+                    return@doesLater
+                }
+                sendRoleMsg(foundRole, message)
+            }
+        }
+        long("roleId") {
+            doesLater { context ->
```
Comment (l1ving) [inline src/main/kotlin/commands/RoleInfoCommand.kt:27]: see above

File: src/main/kotlin/commands/RoleInfoCommand.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                    message.error("Role name not found! Does this role exist?")
+                    return@doesLater
+                }
+                sendRoleMsg(foundRole, message)
+            }
+        }
+    }
+
+    private suspend fun sendRoleMsg(foundRole: Role, message: Message) {
+        if (foundRole.isEveryone) message.warn("Everyone role")
```
Comment (l1ving) [inline src/main/kotlin/commands/RoleInfoCommand.kt:39]: Why is this a warning?

File: src/main/kotlin/commands/RoleInfoCommand.kt [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        // someone make a regex
+        val filteredPermission: String = foundRole.permissions
```
Comment (l1ving) [inline src/main/kotlin/commands/RoleInfoCommand.kt:41]: I already made a pretty formatting for role permissions, look at ChannelCommand

File: src/main/kotlin/commands/RoleInfoCommand.kt [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            .replace(")", "")
+
+        message.channel.send {
+            embed {
+                title = foundRole.name
+                color = foundRole.color
+                field("Permissions", filteredPermission)
+                field("Separate from online members?", foundRole.hoist.toString())
+                field("Position", foundRole.position.toString())
+                field("Mention", foundRole.mention)
```
Comment (l1ving) [inline src/main/kotlin/commands/RoleInfoCommand.kt:56]: Mention should be first

File: src/main/kotlin/commands/RoleInfoCommand.kt [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            .replace("Permissionset", "")
+            .replace("(" ,"")
+            .replace(")", "")
+
+        message.channel.send {
+            embed {
+                title = foundRole.name
+                color = foundRole.color
+                field("Permissions", filteredPermission)
+                field("Separate from online members?", foundRole.hoist.toString())
```
Comment (l1ving) [inline src/main/kotlin/commands/RoleInfoCommand.kt:54]: This is a permission?

**Review by l1ving — CHANGES_REQUESTED** [review]

File: src/main/kotlin/commands/RoleInfoCommand.kt [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    private suspend fun sendRoleMsg(foundRole: Role, message: Message) {
+        val filteredPermission: String = foundRole.permissions.pretty()
+
+        message.channel.send {
+            embed {
+                title = foundRole.name
+                color = foundRole.color
+                field("Mention", foundRole.mention)
+                field("Permissions", "$filteredPermission ${if (foundRole.hoist) ", Separate From Online Members" else ""}")
+                field("Position", foundRole.position.toString())
```
Comment (l1ving) [inline src/main/kotlin/commands/RoleInfoCommand.kt:48]: positions for channels are magic numbers and are entirely worthless, I wouldn't be surprised if roles were the same. Test if it works properly

Comment (sourTaste000) [reply src/main/kotlin/commands/RoleInfoCommand.kt:48]: Not magic numbers
count from bottom role(everyone) to top(mayor)

**Pull request merged**